src/valid/cause.rs

```diff
@@ -29,6 +29,16 @@ impl<E: Display> Display for Cause<E> {
   }
 }
 
+impl From<Cause<&str>> for Cause<String> {
```

Comment:
```suggestion
impl<E: Display> From<Cause<E>> for Cause<E> {
```

This might be a more generic way to implement From.